### PR TITLE
Use updated distribution endpoint

### PR DESCRIPTION
**Pull request by dmfalke**
This PR includes changes to use the updated distribution endpoint for filters in the subsetting tab. The updated endpoint is designed for performance, which is a requirement for subsetting.

One UX compromise is that we cannot do semantic zoom in this context. The front end must always provide range and binning info.

This depends on https://github.com/VEuPathDB/web-components/pull/165 for mouse selection in histograms to work correctly.

### CODE REVIEW

**Comment by dmfalke** [comment]
Note, I plan to merge several commits into one, so that we can easily revert to using the histogram and barplot plugins, if needed.

**Comment by bobular** [comment]
I'm not sure how you plan to do the latter, but sounds good.  I will review
now.

**Comment by dmfalke** [comment]
I think this is ready

**Review by bobular — COMMENTED** [review]
HistogramFilter:
At the moment the year/month/week/day menu is showing up for non-date variables. It would be nice to replace the `"(month)"` shown in the label of BinWidthControl's SliderWidget with the dropdown menu.  A bit of a fiddle but maybe worth it?

Default slider range for date vars could be 1-60?

`22,567 (100%) of 22,567 Participant have data for this variable` seems to have disappeared for HistogramFilter

**Comment by dmfalke** [comment]
Thanks, Bob. All good points. I will address as soon as I can.

**Comment by dmfalke** [comment]
I pushed commits to address these suggestions. See also the relevant commit in https://github.com/VEuPathDB/web-components/pull/165.

File: src/lib/core/components/filter/HistogramFilter.tsx [hunk 1]
```diff
@@ -80,28 +79,32 @@ export function HistogramFilter(props: Props) {
...
     if (variable.type === 'number')
       return {
-        binWidth: variable.binWidthOverride ?? variable.binWidth,
+        binWidth: variable.binWidthOverride ?? variable.binWidth ?? 0.1,
         binWidthTimeUnit: undefined,
         independentAxisRange:
           variable.displayRangeMin != null && variable.displayRangeMax != null
             ? { min: variable.displayRangeMin, max: variable.displayRangeMax }
-            : undefined,
+            : { min: Math.min(0, variable.rangeMin!), max: variable.rangeMax! },
```
Comment (bobular) [inline src/lib/core/components/filter/HistogramFilter.tsx:87]: Should we make `{Date,Number}Variable.range{Min,Max}` mandatory props in types/study.ts?

It seems that if `NumberVariable.rangeMax` is missing, the request to the back end will fall over.

If they are missing from the studies/variables metadata we might get a more meaningful error message?

Same goes for date rangeMin and Max - see below

File: src/lib/core/components/filter/HistogramFilter.tsx [hunk 2]
```diff
@@ -80,28 +79,32 @@ export function HistogramFilter(props: Props) {
...
       independentAxisRange:
         variable.displayRangeMin != null && variable.displayRangeMax != null
-          ? { min: variable.displayRangeMin, max: variable.displayRangeMax }
-          : undefined,
+          ? {
+              min: variable.displayRangeMin + 'T00:00:00',
+              max: variable.displayRangeMax + 'T00:00:00',
+            }
+          : {
+              min: variable.rangeMin! + 'T00:00:00',
```
Comment (bobular) [inline src/lib/core/components/filter/HistogramFilter.tsx:105]: If `variable.rangeMin` is missing, `'T00:00:00'` is likely to be an invalid date for the back end.  

See previous comment above.

**Review by bobular — APPROVED** [review]
Sorry, I meant to start a review but was tired and pressed the wrong button with the previous comments about rangeMin and rangeMax being missing.

If you think it's OK to keep it as-is, I'm all for that.

**Comment by dmfalke** [comment]
I’ll play with making them mandatory. I’m not sure they’ll be populated for
categorical number and date vars, so might require even more factoring.

**Comment by bobular** [comment]
Here's a categorical number var from GEMS1 that does have rangeMin/Max, so
hopefully happy days (but not more than 7)!

{
"63": {
"id": "EUPATH_0010238",
"parentId": "EUPATH_0010016",
"providerLabel": "[\"F4A_DRH_DAYS\"]",
"displayName": "Diarrheal episode duration",
"definition": "Assessed at enrollment for cases",
"vocabulary": [
"1",
"2",
"3",
"4",
"5",
"6",
"7"
],
"displayType": "default",
"displayOrder": 1,
"type": "number",
"dataShape": "categorical",
"units": "",
"precision": 0,
"rangeMin": 1,
"rangeMax": 7,
"binWidth": 0.3,
"distinctValuesCount": 7,
"isTemporal": false,
"isFeatured": false,
"isMergeKey": false,
"isMultiValued": false
}
}

**Comment by dmfalke** [comment]
Shoot, I forgot to squash my commits! Hopefully we wont have to revert anything!

**Comment by dmfalke** [comment]
If we need to revert, I can just revert individual commits. It shouldn't be too bad, since they are colocated in this PR.

**Comment by dmfalke** [comment]
I also directly pushed this commit: https://github.com/VEuPathDB/web-eda/commit/9f4a72796d2594fbbc23232a4d0dd131fabf9063

**Pull request merged**